src/core/api/reactiveSearchv3Api.js

```diff
@@ -23,14 +25,15 @@ function reactiveSearchv3Api(query, settings) {
     settings: parsedSettings,
     query,
   };
-
-
+  if (this.mongodb) {
+    Object.assign(body, { mongodb: getMongoRequest(this.app, this.mongodb) });
+  }
   return this.performFetchRequest({
     method: 'POST',
     path: '_reactivesearch.v3',
     body,
-    headers: getTelemetryHeaders(this.enableTelemetry),
-    isRSAPI: true,
+    headers: getTelemetryHeaders(this.enableTelemetry, !this.mongodb),
+    isRSAPI: !this.mongodb,
```

Comment:
Check it too.

Reply:
https://www.loom.com/share/af739667196a46f3924736a4b311e4df

Reply:
@bietkul 
f5a18e7

Reply:
```suggestion
    isRSAPI: true,
```